### PR TITLE
Fix up CalicoAPIConfig definition to use new resource format

### PR DESCRIPTION
**Pull request by robbrockbank**
Update calico api config def.

### CODE REVIEW

File: lib/apiconfig/load.go [hunk 1]
```diff
@@ -53,8 +53,8 @@ func LoadClientConfigFromBytes(b []byte) (*CalicoAPIConfig, error) {
 	if c.APIVersion != apiv2.GroupVersionCurrent {
 		return nil, errors.New("invalid config file: unknown APIVersion '" + c.APIVersion + "'")
 	}
-	if c.Kind != "calicoApiConfig" {
-		return nil, errors.New("invalid config file: expected kind 'calicoApiConfig', got '" + c.Kind + "'")
+	if c.Kind != KindCalicoAPIConfig {
```
Comment (tmjd) [inline lib/apiconfig/load.go]: This changes the case of the first `c` to `C`, does that matter?

File: lib/apiconfig/apiconfig.go [hunk 2]
```diff
@@ -71,9 +72,9 @@ type KubeConfig struct {
 // TypeMetadata initialised to the current version.
 func NewCalicoAPIConfig() *CalicoAPIConfig {
 	return &CalicoAPIConfig{
-		TypeMetadata: unversioned.TypeMetadata{
-			Kind:       "calicoApiConfig",
-			APIVersion: unversioned.VersionCurrent,
+		TypeMeta: metav1.TypeMeta{
+			Kind:       KindCalicoAPIConfig ,
```
Comment (tmjd) [inline lib/apiconfig/apiconfig.go]: Changed case of first letter.

Comment (robbrockbank) [reply lib/apiconfig/apiconfig.go]: Yup - all our resources are now cap first letter

File: lib/apiconfig/load.go (hunk 1 above)
Comment (robbrockbank) [reply lib/apiconfig/load.go]: nope - as expected as part of the v2 client.

**Pull request merged**